homeassistant/scripts/check_config.py

```diff
@@ -95,9 +95,8 @@ def run(script_args: List) -> int:
     if args.files:
         print(color(C_HEAD, 'yaml files'), '(used /',
               color('red', 'not used') + ')')
-        # Python 3.5 gets a recursive, but not in 3.4
-        for yfn in sorted(glob(os.path.join(config_dir, '*.yaml')) +
-                          glob(os.path.join(config_dir, '*/*.yaml'))):
+        for yfn in sorted(glob(os.path.join(config_dir, '**', '*.yaml'),
```

Comment:
I think the previous code had a bug for Windows here. Probably it didn't get noticed since you have to opt-in to get this executed. From [`fnmatch`](https://docs.python.org/3/library/fnmatch.html#module-fnmatch) docs (which is used by [`glob`](https://docs.python.org/3/library/glob.html#glob.glob)): 

> Note that the filename separator ('/' on Unix) is not special to this module.

So we should be using [`os.sep`](https://docs.python.org/3/library/os.html#os.sep) here. Or in this case the [`os.path.join`](https://docs.python.org/3/library/os.path.html#os.path.join) method.

Reply:
CC @kellerza

Reply:
`recursive` has the unintended consequence of now including `deps/` --> I guess the old one was only one dir deep.

a=old, b=new, same result on Windows and Docker

```
>>> a
['./automation-tower/buttons.yaml', './automation-tower/fountain.yaml', './automation-tower/garage.yaml', './automation-tower/kodi.yaml', './automation-tower/lights.yaml', './automation-tower/location.yaml', './automation-tower/swembad.yaml', './automation-tower/test.yaml', './automations.yaml', './conf3.yaml', './configuration-dev.yaml', './configuration-hal.yaml', './configuration-tower.yaml', './configuration.yaml', './customize.yaml', './entity_registry.yaml', './groups.yaml', './idle/configuration.yaml', './idle/secrets.yaml', './package_sonoff.yaml', './scene.yaml', './script.yaml', './secrets.yaml', './sensors.yaml', './switches.yaml']
>>> b
['./automation-tower/buttons.yaml', './automation-tower/fountain.yaml', './automation-tower/garage.yaml', './automation-tower/kodi.yaml', './automation-tower/lights.yaml', './automation-tower/location.yaml', './automation-tower/swembad.yaml', './automation-tower/test.yaml', './automations.yaml', './conf3.yaml', './configuration-dev.yaml', './configuration-hal.yaml', './configuration-tower.yaml', './configuration.yaml', './customize.yaml', './deps/lib/python3.6/site-packages/scripts/default_broker.yaml', './deps/lib/python3.6/site-packages/scripts/default_client.yaml', './entity_registry.yaml', './groups.yaml', './idle/configuration.yaml', './idle/secrets.yaml', './package_sonoff.yaml', './scene.yaml', './script.yaml', './secrets.yaml', './sensors.yaml', './switches.yaml']
```

Since it is an optional parameter it is not critical, but ideally you want to filter deps:

```
>>> deps = os.path.join(config_dir, 'deps')
>>> [val for val in b if not val.startswith(deps)]
['./automation-tower/buttons.yaml', './automation-tower/fountain.yaml', './automation-tower/garage.yaml', './automation-tower/kodi.yaml', './automation-tower/lights.yaml', './automation-tower/location.yaml', './automation-tower/swembad.yaml', './automation-tower/test.yaml', './automations.yaml', './conf3.yaml', './configuration-dev.yaml', './configuration-hal.yaml', './configuration-tower.yaml', './configuration.yaml', './customize.yaml', './entity_registry.yaml', './groups.yaml', './idle/configuration.yaml', './idle/secrets.yaml', './package_sonoff.yaml', './scene.yaml', './script.yaml', './secrets.yaml', './sensors.yaml', './switches.yaml']
>>> [val for val in b if not val.startswith(deps)] == a
True
```

Reply:
@OttoWinter Windows really does not care about `/` or `\` , even pathlib.Path prints the following on Windows: `WindowsPath('C:/Users...')`